src/types.ts

```diff
@@ -36,3 +36,13 @@ export type TransformEntry = {
 };
 
 export type BuildEntry = BundleEntry | TransformEntry;
+
+export interface BuildConfig {
+  entries?: (BuildEntry | string)[];
+}
+
+export function defineBuildConfig(
+  config: BuildConfig | BuildConfig[]
+): BuildConfig[] {
+  return (Array.isArray(config) ? config : [config]).filter(Boolean)
```

Comment:
Since with obuild each Bundler entry can have an array of inputs (for shared chunk) we don;t need array of top level configs anymore comparing to unbuild.